src/libexpr/eval-settings.cc

```diff
@@ -69,11 +69,9 @@ Strings EvalSettings::getDefaultNixPath() const
         }
     };
 
-    if (!restrictEval && !pureEval) {
```

Comment:
I wasn't aware how problematic this conditional was, but in my mental model it was always initialized to those three elements anyway, because we looked at it from the perspective of the initialization of the settings object anyway, of which we were aware that it was not in a properly loaded state yet. But yeah, it's worse; not even properly initialized, but in effect it doesn't matter which it is.

off-topic: I'd much prefer a good ol' infinite recursion over this stateful mess :D

Reply:
Indeed, an option would be to make all settings (virtual) methods, such that they can compute.

Reply:
That'd be alright, and a call depth limit could replace `tBlackhole` in this comparison.